app/trace/trace_events.ts

```diff
@@ -46,44 +46,60 @@ export type TimeSeriesEvent = TraceEvent & {
 export type TimeSeries = {
   name: string;
   events: TimeSeriesEvent[];
+  unit?: string;
+};
+
+type seriesMetadata = {
+  argKey: string;
+  displayName?: string;
+  unit?: string;
 };
 
 // A list of names of events that contain a timestamp and a value in args.
 //
+// The names from Bazel profiles will be displayed as-is, unless a displayName
+// is provided in the seriesMetadata object.
+//
 // We only render timeseries which are defined in this list. The order of the
 // entries in this list determines the order in which the timeseries are
 // displayed in the trace viewer panel.
 //
-// An example event for Bazel CPU usage looks like this: {"name": "CPU usage
-// (bazel)", ..., "args": {"cpu": 0.84}}
-const TIME_SERIES_EVENT_NAMES_AND_ARG_KEYS_ENTRIES: [string, string][] = [
+// An example event for Bazel CPU usage looks like this:
+// {
+//   "name": "CPU usage (bazel)",
+//   ...,
+//   "args": {"cpu": 0.84}
+// }
+const TIME_SERIES_EVENT_NAMES = new Map<string, seriesMetadata>([
   // Event names/arg keys from Bazel profiles.
   // These are defined by bazel / not controlled by us.
-  ["action count", "action"],
-  ["CPU usage (Bazel)", "cpu"],
-  ["Memory usage (Bazel)", "memory"],
-  ["CPU usage (total)", "system cpu"],
-  ["Memory usage (total)", "system memory"],
-  ["System load average", "load"],
-  ["Network Up usage (total)", "system network up (Mbps)"],
-  ["Network Down usage (total)", "system network down (Mbps)"],
+  ["action count", { argKey: "action" }],
+  ["CPU usage (Bazel)", { argKey: "cpu" }],
+  ["Memory usage (Bazel)", { argKey: "memory", unit: "MB" }],
+  ["CPU usage (total)", { argKey: "system cpu", displayName: "CPU usage (System)" }],
+  ["Memory usage (total)", { argKey: "system memory", unit: "MB" }],
+  ["System load average", { argKey: "load" }],
+  [
+    "Network Up usage (total)",
+    { argKey: "system network up (Mbps)", displayName: "Network Up usage (System)", unit: "Mbps" },
+  ],
+  [
+    "Network Down usage (total)",
+    { argKey: "system network down (Mbps)", displayName: "Network Down usage (System)", unit: "Mbps" },
+  ],
 
   // Event names/arg keys from executor profiles.
-  // These are controlled by us, and defined in
+  // These are controlled by us, {argKey: and defined in
```

Comment:
```suggestion
  // These are controlled by us, and defined in
```

Reply:
not sure how this slipped in there. I blame vim